lib/init.g

```diff
@@ -247,12 +249,22 @@ CallAndInstallPostRestore( function()
     UNBIND_GLOBAL( "DEBUG_LOADING" );
 
     if IsHPCGAP then
+      # In HPC-GAP we want to decide how to handle errors on a per thread
+      # basis. E.g. the break loop can be disabled in a worker thread that
+      # runs tasks.
       BindThreadLocal( "BreakOnError", not GAPInfo.CommandLineOptions.T );
-      BindThreadLocal( "SilentErrors", false );
+      BindThreadLocal(
+        "AlwaysPrintTracebackOnError",
+        GAPInfo.CommandLineOptions.alwaystrace
+      );
+      BindThreadLocal( "SilentNonInteractiveErrors", false );
+      # We store the error messages on a per thread basis to be able to
+      # e.g. access them independently from the main thread.
       BindThreadLocal( "LastErrorMessage", "" );
     else
       ASS_GVAR( "BreakOnError", not GAPInfo.CommandLineOptions.T );
-      ASS_GVAR( "SilentErrors", false );
+      ASS_GVAR( "AlwaysPrintTracebackOnError", GAPInfo.CommandLineOptions.alwaystrace );
+      ASS_GVAR( "SilentNonInteractiveErrors", false );
```

Comment:
Note a change request, just a remark: if the renaming of `SilentErrors` to `SilentNonInteractiveErrors` were in a separate commit, that'd make things even easier to review. But it's fine to keep it now as it is.